file_download.php

```diff
@@ -198,7 +198,12 @@
 	'application/pdf' );
 $t_mime_force_attachment = array( 'application/x-shockwave-flash', 'text/html' );
 
-$t_mime_type = substr( $t_content_type, 0, strpos( $t_content_type, ';' ) );
+# extract mime type from content type
+if( strpos( $t_content_type, ';' ) === false ) {
+        $t_mime_type = $t_content_type;
+} else {
+	$t_mime_type = substr( $t_content_type, 0, strpos( $t_content_type, ';' ) );
+}
```

Comment:
Less efficient that way: 1 or 3 function calls + 1 if and 1 assignment, vs 1 function call and 2 assignments. It's readable enough with the comment.